plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -450,9 +811,16 @@ func (c *CiscoTelemetryMDT) parseContentField(grouper *metric.SeriesGrouper, fie
 			for i, subfield := range row.Fields {
 				if i == 0 { // First subfield contains the index, promote it from value to tag
 					tags[prefix] = decodeTag(subfield)
+                                        //We can have subfield so recursively handle it.
+                                        if (len(row.Fields) == 1) {
+                                            tags["row_number"] = strconv.FormatInt(int64(i), 10)
```

Comment:
This is overwritten seven lines down, so what is the point in setting this here?